packages/perspective/src/js/perspective.parallel.js

```diff
@@ -67,20 +72,31 @@ class WebWorker extends worker {
     }
 
     async register() {
-        let worker;
-        const msg = {cmd: "init"};
         if (typeof WebAssembly === "undefined" || detect_iphone()) {
-            worker = await asmjs_worker();
+            const worker = await asmjs_worker();
+            const msg = {
+                cmd: "init"
+            };
+            for (var key in this._worker) {
+                worker[key] = this._worker[key];
+            }
+            this._worker = worker;
+            this._worker.addEventListener("message", this._handle.bind(this));
+            this._worker.postMessage(msg);
+            this._detect_transferable();
         } else {
-            [worker, msg.buffer] = await Promise.all([override.worker(), override.wasm()]);
-        }
-        for (var key in this._worker) {
-            worker[key] = this._worker[key];
+            const [worker, buffer] = await Promise.all([override.worker(), override.wasm()]);
+            const msg = {
+                cmd: "init"
+            };
+            for (var key in this._worker) {
+                worker[key] = this._worker[key];
+            }
+            this._worker = worker;
+            this._worker.addEventListener("message", this._handle.bind(this));
+            this._worker.postMessage(msg, [buffer]);

```

Comment:
This is [not how transferables work](https://developer.mozilla.org/en-US/docs/Web/API/Transferable) - there is even test coverage for this feature!